server/controllers/message.js

```diff
@@ -0,0 +1,92 @@
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+const Conversation = require("../models/Conversation");
+const Message = require("../models/Message");
+
+const { findConversationQuery } = require("../utils/conversationQueries");
+
+// @route GET /messages/:convoId
+// @desc retrieves all messages by a given conversation id
+// @access Private
+exports.getAllMessages = asyncHandler(async (req, res) => {
+  const {
+    params: { convoId },
+    query: { sort = "createdAt", order = "asc" },
+  } = req;
+  const limit = parseInt(req.query.limit || 30),
+    page = parseInt(req.query.page || 1);
+
+  const messages = await Message.find({
+    conversationId: convoId,
+  })
```

Comment:
Might want to check whether the authenticated user is a part of the conversation provided for security reasons 👍🏼

Reply:
Oh you're right 🤣, that completely slipped my mind when building this route .